functional/FunctionalTest.java

```diff
@@ -3795,6 +3798,155 @@ public static void getDefaultRetention_test() throws Exception {
+        client.setObjectRetention(
+            SetObjectRetentionArgs.builder()
+                .bucket(bucketName)
+                .object(objectName)
+                .config(emptyConfig)
+                .bypassGovernanceMode(true)
+                .build());
+
+      } finally {
+        client.removeObject(
+            RemoveObjectArgs.builder().bucket(bucketName).object(objectName).build());
+        client.removeBucket(RemoveBucketArgs.builder().bucket(bucketName).build());
+      }
+      mintSuccessLog(methodName, null, startTime);
+    } catch (Exception e) {
+      handleException(methodName, null, startTime, e);
+    }
+  }
+
+  /** Test: getObjectRetention(GetObjectRetentionArgs args). */
+  public static void getObjectRetention_test1() throws Exception {
+    String methodName = "getObjectRetention(GetObjectRetentionArgs args)";
+    if (!mintEnv) {
+      System.out.println("Test: with shortened retention period: " + methodName);
+    }
+
+    long startTime = System.currentTimeMillis();
+    String bucketName = getRandomName();
+    String objectName = getRandomName();
+    try {
+      client.makeBucket(MakeBucketArgs.builder().bucket(bucketName).objectLock(true).build());
+      try {
+        try (final InputStream is = new ContentInputStream(1 * KB)) {
+          PutObjectOptions options = new PutObjectOptions(1 * KB, -1);
+          options.setContentType(customContentType);
+          client.putObject(bucketName, objectName, is, options);
+        }
+
+        ZonedDateTime retentionUntil = ZonedDateTime.now(Time.UTC).plusDays(3).withNano(0);
+        Retention expectedConfig = new Retention(RetentionMode.GOVERNANCE, retentionUntil);
+        client.setObjectRetention(
+            SetObjectRetentionArgs.builder()
+                .bucket(bucketName)
+                .object(objectName)
+                .config(expectedConfig)
+                .build());
+
+        Retention config =
+            client.getObjectRetention(
+                GetObjectRetentionArgs.builder().bucket(bucketName).object(objectName).build());
+
+        if (!(config.retainUntilDate().equals(expectedConfig.retainUntilDate()))
+            || (config.mode() != expectedConfig.mode())) {
+          throw new Exception(
+              "[FAILED] Expected: expected duration : "
+                  + expectedConfig.retainUntilDate()
+                  + ", got: "
+                  + config.retainUntilDate()
+                  + " expected mode :"
+                  + expectedConfig.mode()
+                  + ", got: "
+                  + config.mode());
+        }
+
+        // You can perform operations on object versions that are locked in governance mode as if
+        // they were unprotected if you have the s3:bypassGovernanceMode permission. These
+        // operations include deleting an object version, shortening the retention period, or
+        // removing the Object Lock by placing a new lock with empty parameters.
```

Comment:
I don't think this long comment is useful. You could have `// Check shortening retention until period`